File: src/main/java/palei/yurii/imageconverter/core/WebPConverter.java
package palei.yurii.imageconverter.core;

import java.awt.image.BufferedImage;
import java.io.File;
import javax.imageio.IIOImage;
import javax.imageio.ImageIO;
import javax.imageio.ImageWriter;
import javax.imageio.stream.ImageOutputStream;

public class WebPConverter {
  private WebPConverter() {
    // Private constructor to prevent instantiation
  }

  public static void convertToWebP(File inputFile, File outputFile) throws Exception {
    BufferedImage image = ImageIO.read(inputFile);

    var writers = ImageIO.getImageWritersByFormatName("webp");
    if (!writers.hasNext()) {
      throw new IllegalStateException("No WebP writers found");
    }

    ImageWriter writer = writers.next();
    ImageOutputStream ios = ImageIO.createImageOutputStream(outputFile);
    writer.setOutput(ios);

    writer.write(null, new IIOImage(image, null, null), null);

    ios.close();
    writer.dispose();
  }
}

File: src/main/kotlin/palei/yurii/imageconverter/WebPConverter.kt
import java.awt.image.BufferedImage
import java.io.File
import javax.imageio.ImageIO
import javax.imageio.ImageWriter
import javax.imageio.stream.ImageOutputStream

object WebPConverter {
    @Throws(Exception::class)
    fun convertToWebP(inputFile: File, outputFile: File) {
        val image: BufferedImage = ImageIO.read(inputFile)

        val writers: Iterator<ImageWriter> = ImageIO.getImageWritersByFormatName("webp")
        if (!writers.hasNext()) {
            throw IllegalStateException("No WebP writers found")
        }

        val writer: ImageWriter = writers.next()
        val ios: ImageOutputStream = ImageIO.createImageOutputStream(outputFile)
        writer.output = ios

        writer.write(null, javax.imageio.IIOImage(image, null, null), null)

        ios.close()
        writer.dispose()
    }
}